src/flownet/config_parser/_config_parser.py

```diff
@@ -1393,3 +1735,249 @@ def parse_config(
+                f"requires 'min' and 'max' to be defined."
+            )
+
+    # check that mean and stddev is defined for the distributions that need it
+    if (
+        getattr(path_in_config_dict, parameter).distribution == "normal"
+        or getattr(path_in_config_dict, parameter).distribution == "lognormal"
+        or getattr(path_in_config_dict, parameter).distribution == "truncated_normal"
+    ):
+        if not {"mean", "stddev"}.issubset(
+            _check_defined(path_in_config_dict, parameter)
+        ):
+            raise ValueError(
+                f"The '{getattr(path_in_config_dict, parameter).distribution}' distribution for {parameter} "
+                f"requires 'mean' and 'stddev' to be defined."
+            )
+    # check that base is defined for the distributions that need it
+    if getattr(path_in_config_dict, parameter).distribution == "const":
+        if not {"base"}.issubset(_check_defined(path_in_config_dict, parameter)):
+            raise ValueError(
+                f"The '{getattr(path_in_config_dict, parameter).distribution}' distribution for {parameter} "
+                f"requires 'base' to be defined."
+            )
+
+    if getattr(path_in_config_dict, parameter).distribution == "triangular":
+        if {"min", "max", "base", "mean"}.issubset(defined_parameters):
+            raise ValueError(
+                f"The {parameter} has values defined for 'min', 'base', 'mean' and 'max'. "
+                "Only three of them should be defined"
+            )
+        if {"base", "mean", "max"}.issubset(defined_parameters):
+            if (
+                3 * getattr(path_in_config_dict, parameter).mean
+                - getattr(path_in_config_dict, parameter).max
+                - getattr(path_in_config_dict, parameter).base
+                < 0
+            ):
+                raise ValueError(
+                    f"The 'base', 'mean' and 'max' values for {parameter} will "
+                    "give negative 'min' in the triangular distribution."
+                )
+        if not (
+            {"min", "base", "max"}.issubset(defined_parameters)
+            or {"min", "mean", "max"}.issubset(defined_parameters)
+            or {"min", "base", "mean"}.issubset(defined_parameters)
+            or {"max", "base", "mean"}.issubset(defined_parameters)
+        ):
+            raise ValueError(
+                "The triangular distribution needs 'min', 'base', and 'max' to be defined. "
+                "Alternatively two of the former in addition to the 'mean'."
+            )
+
+
+def _check_defined(path_in_config_dict: dict, parameter: str):
+    """
+
+    Args:
+        path_in_config_dict (str): a location in the config schema dictionary
+        parameter (str): a parameter/dictionary found at the given location
+
+    Returns:
+        Nothing, raises ValueErrors if something is wrong
```

Comment:
Incorrect, it does return something and does not throw a ValueError

Reply:
Copy/paste always dangerous....